alchemist/alchemist-fxui/src/main/resources/it/unibo/alchemist/gui/keybinds.json

```diff
@@ -0,0 +1,13 @@
+{
+  "MODIFIER_CONTROL": "CONTROL",
+  "MODIFIER_SHIFT": "SHIFT",
+  "PAN_NORTH": "W",
+  "PAN_SOUTH": "S",
+  "PAN_EAST": "D",
+  "PAN_WEST": "A",
+  "DELETE": "DELETE",
+  "MOVE": "M",
+  "PLAY_AND_PAUSE": "P",
+  "ONE_STEP": "PERIOD",
+  "EDIT": "UNDEFINED"
```

Comment:
I guess this is because it's unsupported yet, right?

Reply:
Yes, although with the way I implemented it, the "EDIT" action would still show up in the GUI for rebinding keys, so functionally it doesn't matter whether it's specified as "UNDEFINED" or not.

Reply:
Implication?

Reply:
I understand there will be an entry in the keybinding gui showing an "Edit" action. I do not understand if "undefined" has a special meaning, or if it is a placeholder and anything but a valid keyboard button would work.
What if I write "foo" in place of "UNDEFINED"?

Reply:
This file is a Map<[ActionFromKey](https://github.com/Vuksaa/Alchemist/blob/develop/alchemist/alchemist-fxui/src/main/kotlin/it/unibo/alchemist/input/Keybinds.kt#L25), [KeyCode](https://docs.oracle.com/javase/8/javafx/api/javafx/scene/input/KeyCode.html)> serialized with Gson, so the string "UNDEFINED" equates to the [UNDEFINED](https://docs.oracle.com/javase/8/javafx/api/javafx/scene/input/KeyCode.html#UNDEFINED) keycode.
If it were "foo" I guess Gson would crash, I haven't implemented any checks.

On a side note, I'll also split the classes in [Keybinds.kt](https://github.com/Vuksaa/Alchemist/blob/develop/alchemist/alchemist-fxui/src/main/kotlin/it/unibo/alchemist/input/Keybinds.kt) and [Keybinder.kt](https://github.com/Vuksaa/Alchemist/blob/develop/alchemist/alchemist-fxui/src/main/kotlin/it/unibo/alchemist/boundary/Keybinder.kt) in separate files.

Reply:
Understood!